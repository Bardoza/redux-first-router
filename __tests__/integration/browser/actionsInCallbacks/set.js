import createTest, { resetBrowser } from '../../../../__helpers__/createTest'
import { set } from '../../../../src/actions'

beforeEach(resetBrowser)

createTest('set before enter throws error', {
  FIRST: '/',
  SECOND: {
    path: '/second',
    beforeEnter: () => set({ query: { hell: 'yea' } }),
    thunk: function() {}
  }
}, {
  testBrowser: true,
  wallabyErrors: false
})

createTest('set after enter', {
  FIRST: '/',
  SECOND: {
    path: '/second',
    thunk: ({ query }) => {
      if (query.hell) return
      return set({ query: { hell: 'yea' } })
    },
    onComplete: function() {}
  }
}, { testBrowser: true })

createTest('set before enter on load throws error', {
  FIRST: {
    path: '/',
    beforeEnter: () => set({ query: { hell: 'yea' } }),
    thunk: function() {}
  }
}, {
  testBrowser: true,
  wallabyErrors: false
})

createTest('set after enter on load', {
  FIRST: {
    path: '/',
    thunk: ({ query }) => {
      if (query.hell) return
      return set({ query: { hell: 'yea' } })
    },
    onComplete: function() {}
  }
}, { testBrowser: true })

createTest('set in pathlessRoute', {
  FIRST: '/',
  PATHLESS: {
    thunk: () => set({ query: { hell: 'yea' } })
  }
}, { testBrowser: true }, [{ type: 'PATHLESS' }])
