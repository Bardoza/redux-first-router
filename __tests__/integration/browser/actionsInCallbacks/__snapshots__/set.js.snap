// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set after enter on load: 1 - routes - FIRST - onComplete 1`] = `1`;

exports[`set after enter on load: 1 - routes - FIRST - thunk 1`] = `2`;

exports[`set after enter on load: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`set after enter on load: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`set after enter on load: firstRoute - /first - 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "hell=yea",
      "url": "/?hell=yea",
    },
    "params": Object {},
    "query": Object {
      "hell": "yea",
    },
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set after enter on load: firstRoute - /first - 1 - history_index 1`] = `0`;

exports[`set after enter on load: firstRoute - /first - 1 - initial_state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@redux/INIT",
}
`;

exports[`set after enter on load: firstRoute - /first - 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "hell=yea",
          "url": "/?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "scene": "",
    "search": "hell=yea",
    "status": 200,
    "url": "/?hell=yea",
  },
  "params": Object {},
  "query": Object {
    "hell": "yea",
  },
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`set after enter on load: firstRoute - /first - 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/?hell=yea",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set after enter on load: firstRoute - /first - 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "hell=yea",
          "url": "/?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {
      "hell": "yea",
    },
    "scene": "",
    "search": "hell=yea",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/?hell=yea",
  },
  "title": "FIRST",
}
`;

exports[`set after enter on load: firstRoute - /first - 1 - title 1`] = `"FIRST"`;

exports[`set after enter on load: firstRoute - /first - 1 - windowLocation 1`] = `"/?hell=yea"`;

exports[`set after enter: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/second",
      "scene": "",
      "search": "hell=yea",
      "url": "/second?hell=yea",
    },
    "params": Object {},
    "query": Object {
      "hell": "yea",
    },
    "state": Object {},
    "type": "SECOND",
  },
]
`;

exports[`set after enter: 1 - history_index 1`] = `1`;

exports[`set after enter: 1 - response 1`] = `
Object {
  "basename": "",
  "commit": [Function],
  "hash": "",
  "location": Object {
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "hell=yea",
          "url": "/second?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "index": 1,
    "key": "345678",
    "kind": "set",
    "length": 2,
    "n": 1,
    "pathname": "/second",
    "scene": "",
    "search": "hell=yea",
    "url": "/second?hell=yea",
  },
  "params": Object {},
  "query": Object {
    "hell": "yea",
  },
  "state": Object {},
  "type": "SECOND",
}
`;

exports[`set after enter: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`set after enter: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`set after enter: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`set after enter: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`set after enter: 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/",
      Object {},
      "345678",
    ],
    Array [
      "/second?hell=yea",
      Object {},
      "345678",
    ],
  ],
  "index": 1,
}
`;

exports[`set after enter: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "hell=yea",
          "url": "/second?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/",
        "scene": "",
        "search": "",
        "url": "/",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {
      "hell": "yea",
    },
    "scene": "",
    "search": "hell=yea",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second?hell=yea",
  },
  "title": "SECOND",
}
`;

exports[`set after enter: 1 - title 1`] = `"SECOND"`;

exports[`set after enter: 1 - windowLocation 1`] = `"/second?hell=yea"`;

exports[`set after enter: firstRoute - /first - 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set after enter: firstRoute - /first - 1 - history_index 1`] = `0`;

exports[`set after enter: firstRoute - /first - 1 - initial_state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@redux/INIT",
}
`;

exports[`set after enter: firstRoute - /first - 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`set after enter: firstRoute - /first - 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set after enter: firstRoute - /first - 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "FIRST",
}
`;

exports[`set after enter: firstRoute - /first - 1 - title 1`] = `"FIRST"`;

exports[`set after enter: firstRoute - /first - 1 - windowLocation 1`] = `"/"`;

exports[`set before enter on load throws error: 1 - routes - FIRST - beforeEnter 1`] = `1`;

exports[`set before enter on load throws error: 1 - routes - FIRST - thunk 1`] = `0`;

exports[`set before enter on load throws error: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`set before enter on load throws error: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - history_index 1`] = `0`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - initial_state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@redux/INIT",
}
`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - response 1`] = `
Object {
  "error": [Error: [rudy] "set" cant be used before enter -- redirect instead],
  "type": "@@rudy/CALL_HISTORY_ERROR",
}
`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - sessionStorage 1`] = `null`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "error": [Error: [rudy] "set" cant be used before enter -- redirect instead],
    "errorType": "@@rudy/CALL_HISTORY_ERROR",
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@rudy/CALL_HISTORY_ERROR",
}
`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - title 1`] = `"@@rudy/CALL_HISTORY_ERROR"`;

exports[`set before enter on load throws error: firstRoute - /first - 1 - windowLocation 1`] = `"/"`;

exports[`set before enter throws error: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set before enter throws error: 1 - history_index 1`] = `0`;

exports[`set before enter throws error: 1 - response 1`] = `
Object {
  "error": [Error: [rudy] "set" cant be used before enter -- redirect instead],
  "type": "@@rudy/CALL_HISTORY_ERROR",
}
`;

exports[`set before enter throws error: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`set before enter throws error: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`set before enter throws error: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`set before enter throws error: 1 - routes - SECOND - thunk 1`] = `0`;

exports[`set before enter throws error: 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set before enter throws error: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "error": [Error: [rudy] "set" cant be used before enter -- redirect instead],
    "errorType": "@@rudy/CALL_HISTORY_ERROR",
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@rudy/CALL_HISTORY_ERROR",
}
`;

exports[`set before enter throws error: 1 - title 1`] = `"@@rudy/CALL_HISTORY_ERROR"`;

exports[`set before enter throws error: 1 - windowLocation 1`] = `"/"`;

exports[`set before enter throws error: firstRoute - /first - 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set before enter throws error: firstRoute - /first - 1 - history_index 1`] = `0`;

exports[`set before enter throws error: firstRoute - /first - 1 - initial_state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@redux/INIT",
}
`;

exports[`set before enter throws error: firstRoute - /first - 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`set before enter throws error: firstRoute - /first - 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set before enter throws error: firstRoute - /first - 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "FIRST",
}
`;

exports[`set before enter throws error: firstRoute - /first - 1 - title 1`] = `"FIRST"`;

exports[`set before enter throws error: firstRoute - /first - 1 - windowLocation 1`] = `"/"`;

exports[`set in pathlessRoute: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "hell=yea",
      "url": "/?hell=yea",
    },
    "params": Object {},
    "query": Object {
      "hell": "yea",
    },
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set in pathlessRoute: 1 - history_index 1`] = `0`;

exports[`set in pathlessRoute: 1 - response 1`] = `
Object {
  "basename": "",
  "commit": [Function],
  "hash": "",
  "location": Object {
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "hell=yea",
          "url": "/?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "index": 0,
    "key": "345678",
    "kind": "set",
    "length": 1,
    "n": 1,
    "pathname": "/",
    "scene": "",
    "search": "hell=yea",
    "url": "/?hell=yea",
  },
  "params": Object {},
  "query": Object {
    "hell": "yea",
  },
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`set in pathlessRoute: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`set in pathlessRoute: 1 - routes - PATHLESS - thunk 1`] = `1`;

exports[`set in pathlessRoute: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`set in pathlessRoute: 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/?hell=yea",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set in pathlessRoute: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "hell=yea",
          "url": "/?hell=yea",
        },
        "params": Object {},
        "query": Object {
          "hell": "yea",
        },
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {
      "hell": "yea",
    },
    "scene": "",
    "search": "hell=yea",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/?hell=yea",
  },
  "title": "FIRST",
}
`;

exports[`set in pathlessRoute: 1 - title 1`] = `"FIRST"`;

exports[`set in pathlessRoute: 1 - windowLocation 1`] = `"/?hell=yea"`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "location": Object {
      "key": "345678",
      "pathname": "/",
      "scene": "",
      "search": "",
      "url": "/",
    },
    "params": Object {},
    "query": Object {},
    "state": Object {},
    "type": "FIRST",
  },
]
`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - history_index 1`] = `0`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - initial_state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "init",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "@@redux/INIT",
}
`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - sessionStorage 1`] = `
Object {
  "entries": Array [
    Array [
      "/",
      Object {},
      "345678",
    ],
  ],
  "index": 0,
}
`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/",
          "scene": "",
          "search": "",
          "url": "/",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "key": "345678",
    "kind": "load",
    "length": 1,
    "n": 1,
    "params": Object {},
    "pathname": "/",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": -1,
        "key": "",
        "pathname": "",
        "scene": "",
        "search": "",
        "url": "",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/",
  },
  "title": "FIRST",
}
`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - title 1`] = `"FIRST"`;

exports[`set in pathlessRoute: firstRoute - /first - 1 - windowLocation 1`] = `"/"`;
